src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/ProviderLocationsControllerTests.cs

```diff
@@ -0,0 +1,113 @@
+                Email = "test2@test.com",
+                Phone = "1234567892"
+            };
+            response.Add(ProviderLocation1);
+            response.Add(ProviderLocation2);
+
+            _mediator = new Mock<IMediator>();
+            _mediator.Setup(x => x.Send(It.IsAny<GetProviderLocationQuery>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(() => new GetProviderLocationQueryResult
+                {
+                    ProviderLocations = response
+                });
+
+            _controller = new ProviderLocationsController(_mediator.Object,  _logger.Object)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = new DefaultHttpContext() { User = user },
+                },
+                TempData = Mock.Of<ITempDataDictionary>()
+            };
+        }
+
+        [Test]
+        public async Task ProviderLocationsController_ViewProviderLocations_ReturnsValidResponse()
+        {
+            var result = await _controller.ViewProviderLocations();
+
+            var viewResult = result as ViewResult;
+            viewResult.Should().NotBeNull();
+            viewResult.ViewName.Should().Contain("ViewProviderLocations.cshtml");
+            viewResult.Model.Should().NotBeNull();
+            var model = viewResult.Model as ProviderLocationListViewModel;
+            model.Should().NotBeNull();
+            model.ProviderLocations.Should().NotBeNull();
+        }
+
+        [Test]
+        public async Task ProviderLocationsController_ViewProviderLocations_ReturnsNoProviderLocationData()
+        {
+            _mediator.Setup(x => x.Send(It.IsAny<GetProviderLocationQuery>(), It.IsAny<CancellationToken>()))
+                     .ReturnsAsync(() => null);
+            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[]{ new Claim(ProviderClaims.ProviderUkprn,"111"),}, "mock"));
+            
+            _controller = new ProviderLocationsController(_mediator.Object, _logger.Object)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = new DefaultHttpContext() { User = user },
+                },
+                TempData = Mock.Of<ITempDataDictionary>()
+            };
+            var result = await _controller.ViewProviderLocations();
+
+            var viewResult = result as ViewResult;
+            viewResult.Should().NotBeNull();
+            viewResult.ViewName.Should().Contain("ViewProviderLocations.cshtml");
+            viewResult.Model.Should().NotBeNull();
+            var model = viewResult.Model as ProviderLocationListViewModel;
+            model.Should().NotBeNull();
+            model.ProviderLocations.Should().BeEmpty();
+            _logger.Verify(x => x.Log(LogLevel.Information, It.IsAny<EventId>(), It.IsAny<It.IsAnyType>(), It.IsAny<Exception>(), It.IsAny<Func<It.IsAnyType, Exception, string>>()), Times.Exactly(2));
```

Comment:
no need for logger verification every where... if we add more log statements the test will fail.